jupyter_notebook/notebookapp.py

```diff
@@ -261,6 +263,22 @@ def init_handlers(self, settings):
         new_handlers.append((r'(.*)', Template404))
         return new_handlers
 
+    def _create_widget_shim_handlers(self):
+        """Creates a handler that shims the loading of widgets
+        JS files for the old /static/widgets/js/*.js path."""
+        class DeprecationHandler(IPythonHandler):
```

Comment:
This class can be defined normally in the module. It doesn't need to be in a closure.